### PR TITLE
libhybris: fix various compile time warnings

### PR DESCRIPTION
**Pull request by vicamo**
Closes: #304
Signed-off-by: You-Sheng Yang <vicamo@gmail.com>

### CODE REVIEW

File: hybris/common/dlfcn.c [hunk 1]
```diff
@@ -36,7 +36,7 @@ int   hybris_dlclose(void *handle)
 }
 
 
-char *hybris_dlerror(void)
+const char *hybris_dlerror(void)
```
Comment (vicamo) [inline hybris/common/dlfcn.c]: `android_dlerror()` returns __const char*__.

File: hybris/common/jb/dlfcn.c [hunk 2]
```diff
@@ -13,10 +13,10 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-#define _GNU_SOURCE
```
Comment (vicamo) [inline hybris/common/jb/dlfcn.c]: Has been defined in CFLAGS in Makefile.am.

**Comment by vicamo** [comment]
Still some left behind, such as checking return values for `write()`, and `pthread_attr_getstackaddr()` has been declared deprecated, etc.

**Comment by morphis** [comment]
LGTM

**Pull request merged**